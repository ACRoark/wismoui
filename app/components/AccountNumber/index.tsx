import * as React from 'react';

import formatAccountNumber from './formatAccountNumber';

interface IAccountNumberProps {
  value: string;
}

const AccountNumber: React.FC<IAccountNumberProps> = (props: IAccountNumberProps): React.ReactElement => {
  const { value } = props;

  const formattedAccountNumber = formatAccountNumber(value);

  return <span>{formattedAccountNumber}</span>;
};

export default AccountNumber;
