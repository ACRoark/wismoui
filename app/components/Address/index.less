@import (reference) '../../styles/themes/index.less';

@address-cls-prefix: ~'@{dte-prefix}-address';

.@{address-cls-prefix} {
  display: inline;
  font-style: normal;
  line-height: 1.375rem;
  white-space: pre-line;
}
