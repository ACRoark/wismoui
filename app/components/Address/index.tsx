import React from 'react';

import AddressFormat from 'components/Address/AddressFormat';
import IAddress from 'types/IAddress';

import formatAddress from './formatAddress';

import './index.less';

interface IAddressProps {
  address: IAddress;
  format?: AddressFormat;
}

const Address: React.FC<IAddressProps> = (props: IAddressProps): React.ReactElement => {
  const { address, format = 'block' } = props;

  return <address className="dte-wismo-address">{formatAddress(address, format)}</address>;
};

export default Address;
