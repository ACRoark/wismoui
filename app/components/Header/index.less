@import (reference) '../../styles/themes/index.less';

@desktop: ~'(min-width: @{desktop-min-width})';

.@{dte-prefix} {
  &-header-background {
    background-color: @dte-background-brand-blue;
    color: @dte-content-primary-inverse;
    height: 96px;
    top: 0;
    width: 100%;

    @media @mobile {
      height: 90px;
    }
  }

  &-header {
    @media @desktop {
      align-items: center;
      display: flex;
      height: 96px;
      justify-content: center;
    }

    @media @mobile {
      display: none;
    }
  }

  &-header-link {
    font-size: @dte-wismo-font-size-0;
    line-height: 2rem;
    position: absolute;
    right: 40px;
    text-align: right;

    > a {
      color: white;
      text-decoration: none;

      &:active,
      &:hover,
      &:link,
      &:visited {
        color: white;
        text-decoration: none;
      }
    }
  }

  &-header-logo {
    color: @dte-white;
    height: 32px;
    width: 93px;
  }

  &-header-mobile {
    align-items: center;
    display: flex;
    flex-direction: column;

    &-background-overlay {
      display: none;
    }

    &-center {
      margin-top: 16px;

      .@{dte-prefix}-header-page-title {
        font-size: @dte-wismo-font-size-1;
        height: 25px;
        left: auto;
        line-height: 1.5625rem;
        margin: 13px 0 12px;
        position: static;
        text-align: center;
      }
    }

    &-hamburger {
      background-color: transparent;
      border: none;
      left: 0;
      line-height: 0;
      margin-left: 24px;
      margin-top: 18px;
      outline: none;
      padding: 0;
      position: absolute;
    }

    &-logo {
      color: @dte-white;
      display: flex;
      height: 24px;
      justify-content: center;
      line-height: 0;

      .@{dte-prefix}-header-logo {
        color: @dte-white;
        height: 100%;
      }
    }

    &-menu {
      background-color: @dte-gray-200;
      color: @dte-blue-light;
      display: block;
      font-family: @dte-wismo-font-family;
      font-size: @dte-wismo-font-size-1;
      font-weight: 600;
      height: 100%;
      left: 0;
      position: absolute;
      top: 0;
      width: 281px;
      z-index: 1;

      &-background-overlay {
        height: 100%;
        position: absolute;
        width: 100%;
      }

      &-button {
        background-color: transparent;
        border: none;
        color: @dte-gray-500;
        display: block;
        margin-bottom: 46.5px;
        outline: none;
        padding-top: 31px;
      }

      &-submenu {
        padding-left: 23px;

        > a {
          color: @dte-blue-light;
          text-decoration: none;

          &:active,
          &:focus,
          &:hover,
          &:visited {
            color: @dte-blue-light;
            text-decoration: none;
          }
        }

        &-bottom {
          padding-top: 18px;
        }

        &-top {
          background-color: @dte-white;
          padding-bottom: 18px;
        }
      }
    }

    @media @desktop {
      display: none;
    }
  }

  &-header-page-title {
    font-size: @dte-wismo-font-size-3;
    height: 30px;
    left: 40px;
    line-height: 2rem;
    position: absolute;
  }
}
