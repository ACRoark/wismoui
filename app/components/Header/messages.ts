import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.header';

export default defineMessages({
  link: {
    id: `${scope}.link`,
  },
  serviceRequestsLink: {
    id: `${scope}.service-requests-link`,
  },
  title: {
    id: `${scope}.title`,
  },
});
