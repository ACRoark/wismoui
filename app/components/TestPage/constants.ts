import IOrderRequest from 'types/IOrderRequest';
import IProduct from '../../types/IProduct';

export const fakeProductList: IProduct[] = [
  {
    productType: 'GCR_A_H',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1',
    serviceDate: '2020-04-22',
  },
  {
    productType: 'EFR_D1_1',
    serviceDate: '2020-04-22',
  },
];

export const fakeServiceRequest: IOrderRequest = {
  accountNumber: '111122233334',
  address: {
    city: 'Bloomfield',
    line1: '123 Main',
    line2: null,
    state: 'MI',
    zip: '48301',
  },
  contactPhoneNumber: '7348675309',
  orderRequestStatusUpdates: [
    {
      createdAtDate: '20200319',
      createdAtTime: '131530',
      orderRequestStatus: 'REQUESTED',
    },
  ],
  orderRequestType: 'MIMO_START',
  premiseId: 'premise123',
  products: [
    {
      productType: 'GCR_A_H',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1',
      serviceDate: '2020-04-22',
    },
    {
      productType: 'EFR_D1_1',
      serviceDate: '2020-04-22',
    },
  ],
  serviceOrderEvents: [
    {
      appointment: {
        date: '2020-04-17',
        slotType: 'AM',
      },
      latestStatus: {
          createdAtDate: '20200305',
          createdAtTime: '223443',
          serviceOrderEventStatus: 'COMPLETED',
        },
        product: 'EFR_D1',
      serviceOrderCategory: null,
      serviceOrderId: '123456789',
    },
  ],
  verificationEvents: [
    {
      verificationCategory: 'R975',
      verificationEventId: '123123',
      verificationEventStatusUpdates: [
        {
          verificationCode: 'AD',
          verificationEventStatus: 'COMPLETED',
          createdAtDate: '20200319',
          createdAtTime: '131530',
        },
      ],
    },
  ],
  wantDate: '2020-04-22',
};

export const fakeStartServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_START',
};

export const fakeStopServiceRequest: IOrderRequest = {
  ...fakeServiceRequest,
  orderRequestType: 'MIMO_STOP',
};
