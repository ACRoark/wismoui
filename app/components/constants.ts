import IApplicationConfig from 'types/IApplicationConfig';

export const PageTitles = Object.freeze({
  ORDER_STATUS_PAGE: 'Track Your Order | DTE Energy',
  SEARCH_PAGE: 'Order Tracker | DTE Energy',
  SUMMARY_PAGE: 'Track Your Order | DTE Energy',
});

export const testConfig: IApplicationConfig = {
  flags: {
    canChangeLanguage: false,
    developerMode: false,
  },
  gigyaSiteKey: 'foo',
  googleTagManagerConfig: {
    auth: 'p3zyt1YCX84XB82sOrjMGw', // '',
    id: 'GTM-K2WP588', // 'GTM-PBB6QZ4',
    preview: 'env-1', // '',
  },
  loading: true,
  recaptchaSitekey: '6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI',
  urls: {
    assistanceURL: 'https://www.dteenergy.com',
    backToDTE: 'https://www.dteenergy.com',
    backToMIMO: 'https://www.dteenergy.com',
    contactUs: 'https://www.contact.us.com',
    createAccount: 'https://www.create.account.com',
    decryptTokenApi: 'https://web-test.dteenergy.com/api/decryptToken?detokenizeValue=',
    documentSubmission: 'https://web-test.dteenergy.com/wps/wcm/connect/dte-web/quicklinks/web-form?detokenizeValue=',
    dynatraceScript:
      'https://dynatrace.dteenergy.com:9999/jstag/managed/98834406-1ae1-4257-992c-5b3c5f828584/968d8c7df07fe18c_complete.js',
    editPhoneNumber: 'https://www.edit.phone.number.com',
    editWantDate: 'https://www.edit.service.date.com',
    encryptTokenApi: 'https://web-test.dteenergy.com/api/encryptToken',
    findOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}?name={name}',
    getOrder: 'https://firebase.or.jsonserver/orders/{orderNumber}',
    guestPay: 'https://www.guest.pay.com',
    headerLogo: 'https://www.header.logo.com',
    privacyPolicy: 'https://www.privacy.policy.com',
    searchOrders: 'https://firebase.or.jsonserver/orders',
    signIn: 'https://www.sign.in.com',
    termsAndConditions: 'https://www.terms.and.conditions.com',
  },
};
