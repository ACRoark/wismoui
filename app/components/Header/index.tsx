/* eslint-disable global-require */
import React, { FC, ReactElement, useState } from 'react';
import { FormattedMessage } from 'react-intl';

import Hyperlink from 'components/Hyperlink';
import useConfig from 'hooks/useConfig';
import CloseIcon from 'images/CloseIcon';
import DteLogo from 'images/DteLogo';
import HamburgerMenu from 'images/HamburgerMenu';

import './index.less';
import messages from './messages';

const Header: FC = (): ReactElement => {
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
  const { urls } = useConfig();

  const toggleMobileMenu = (): void => {
    setIsMobileMenuOpen(!isMobileMenuOpen);
  };

  return (
    <header className="dte-wismo-header-background">
      <div className="dte-wismo-header-mobile">
        <button
          className="dte-wismo-header-mobile-hamburger"
          data-testid="open-mobile-menu"
          onClick={toggleMobileMenu}
          type="button"
        >
          <HamburgerMenu title="menu" />
        </button>
        {isMobileMenuOpen && (
          <>
            <div
              aria-label="overlay"
              className="dte-wismo-header-mobile-menu-background-overlay"
              data-testid="close-mobile-menu-overlay"
              onClick={toggleMobileMenu}
              onKeyPress={toggleMobileMenu}
              role="button"
              tabIndex={0}
            />
            <div className="dte-wismo-header-mobile-menu" data-testid="mobile-menu">
              <div className="dte-wismo-header-mobile-menu-submenu dte-wismo-header-mobile-menu-submenu-top">
                <button
                  className="dte-wismo-header-mobile-menu-button"
                  data-testid="close-mobile-menu"
                  onClick={toggleMobileMenu}
                  type="button"
                >
                  <CloseIcon title="Close mobile menu" />
                </button>
                <Hyperlink href="/">
                  <FormattedMessage {...messages.serviceRequestsLink} />
                </Hyperlink>
              </div>
              <div className="dte-wismo-header-mobile-menu-submenu dte-wismo-header-mobile-menu-submenu-bottom">
                <Hyperlink href={urls.backToDTE} target="_self">
                  <FormattedMessage {...messages.link} />
                </Hyperlink>
              </div>
            </div>
          </>
        )}
        <div className="dte-wismo-header-mobile-center">
          <div className="dte-wismo-header-mobile-logo">
            <a href={urls.headerLogo} title="DTE Energy">
              <DteLogo className="dte-wismo-header-logo" title="DTE Energy" />
            </a>
          </div>
          <div className="dte-wismo-header-page-title">
            <FormattedMessage {...messages.title} />
          </div>
        </div>
      </div>

      <div className="dte-wismo-header">
        <div className="dte-wismo-header-page-title">
          <FormattedMessage {...messages.title} />
        </div>
        <div>
          <a href={urls.headerLogo} title="DTE Energy">
            <DteLogo className="dte-wismo-header-logo" title="DTE Energy" />
          </a>
        </div>
        <div className="dte-wismo-header-link">
          <Hyperlink href={urls.backToDTE} target="_self">
            <FormattedMessage {...messages.link} />
          </Hyperlink>
        </div>
      </div>
    </header>
  );
};

export default Header;
