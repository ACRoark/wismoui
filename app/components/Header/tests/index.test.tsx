import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { RawIntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

import { SUPPORTED_LOCALES } from 'locales';

import { testConfig } from 'components/constants';
import useConfig from 'hooks/useConfig';
import { createMockStore, createSnapshotWithIntl, shallowWithIntl } from 'testing/utils';
import createIntlShape from 'testing/utils/createIntlShape';

import Header from '..';

jest.mock('hooks/useConfig');

const mockConfig = useConfig as jest.MockedFunction<typeof useConfig>;

describe('<Header />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);
  const store = createMockStore();

  mockConfig.mockReturnValue({ ...testConfig });

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered in ${language}`, (): void => {
      const spy = jest.spyOn(global.console, 'error');

      shallowWithIntl(<Provider store={store}><Header /></Provider>, language);

      expect(spy).not.toHaveBeenCalled();
    });

    it(`should render correctly when the current language is ${language}`, (): void => {
      const tree = createSnapshotWithIntl(<Provider store={store}><Header /></Provider>, language);

      expect(tree).toMatchSnapshot();
    });
  });

  it('should render correctly when opening and closing the mobile menu', (): void => {
    const intl = createIntlShape('en');

    render(<Router><RawIntlProvider value={intl}><Provider store={store}><Header /></Provider></RawIntlProvider></Router>);

    expect(screen.queryByTestId('mobile-menu')).toBeNull();

    fireEvent.click(screen.getByTestId('open-mobile-menu'));

    expect(screen.getByTestId('mobile-menu')).toBeTruthy();

    fireEvent.click(screen.getByTestId('close-mobile-menu'));

    expect(screen.queryByTestId('mobile-menu')).toBeNull();
  });

  it('should render correctly when opening and closing the mobile menu using the overlay', (): void => {
    const intl = createIntlShape('en');

    render(<Router><RawIntlProvider value={intl}><Header /></RawIntlProvider></Router>);

    expect(screen.queryByTestId('mobile-menu')).toBeNull();

    fireEvent.click(screen.getByTestId('open-mobile-menu'));

    expect(screen.getByTestId('mobile-menu')).toBeTruthy();

    fireEvent.click(screen.getByTestId('close-mobile-menu-overlay'));

    expect(screen.queryByTestId('mobile-menu')).toBeNull();
  });
});
