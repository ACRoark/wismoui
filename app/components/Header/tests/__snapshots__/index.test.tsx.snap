// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly when the current language is en 1`] = `
<header
  className="dte-wismo-header-background"
>
  <div
    className="dte-wismo-header-mobile"
  >
    <button
      className="dte-wismo-header-mobile-hamburger"
      data-testid="open-mobile-menu"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 32 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          menu
        </title>
        <path
          clipRule="evenodd"
          d="M0 1C0 0.447715 0.447715 0 1 0H31C31.5523 0 32 0.447715 32 1C32 1.55228 31.5523 2 31 2H1C0.447715 2 0 1.55228 0 1ZM0 10C0 9.44771 0.447715 9 1 9H31C31.5523 9 32 9.44771 32 10C32 10.5523 31.5523 11 31 11H1C0.447715 11 0 10.5523 0 10ZM0 19C0 18.4477 0.447715 18 1 18H31C31.5523 18 32 18.4477 32 19C32 19.5523 31.5523 20 31 20H1C0.447715 20 0 19.5523 0 19Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </button>
    <div
      className="dte-wismo-header-mobile-center"
    >
      <div
        className="dte-wismo-header-mobile-logo"
      >
        <a
          href="https://www.header.logo.com"
          title="DTE Energy"
        >
          <svg
            className="dte-wismo-header-logo"
            fill="none"
            height="32"
            viewBox="0 0 93 32"
            width="93"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              DTE Energy
            </title>
            <path
              d="M62.0682 7.78728H53.1416V32H41.5539V7.78728H32.6094V0.0179513H62.0682V7.78728ZM92.7523 7.66162V0H66.4833V24.3133C66.4833 28.5103 69.8696 31.9102 74.0418 31.9102H78.9605H92.888V24.2127H79.4035C78.4283 24.2127 77.6353 23.4157 77.6353 22.4355V19.2761H92.145V12.1853H77.6353V7.66162H92.7523ZM19.6178 31.9749H0V0.0502637H19.5535C24.7223 0.0502637 28.9087 4.26164 28.9087 9.45316V22.6366C28.9087 27.7922 24.7509 31.9749 19.6178 31.9749ZM18.1961 21.8826V10.0061C18.1961 8.65253 17.1031 7.5575 15.76 7.5575H11.4699V24.3312H15.76C17.1031 24.3312 18.1961 23.2362 18.1961 21.8826Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
      <div
        className="dte-wismo-header-page-title"
      >
        Order Tracker
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-header"
  >
    <div
      className="dte-wismo-header-page-title"
    >
      Order Tracker
    </div>
    <div>
      <a
        href="https://www.header.logo.com"
        title="DTE Energy"
      >
        <svg
          className="dte-wismo-header-logo"
          fill="none"
          height="32"
          viewBox="0 0 93 32"
          width="93"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            DTE Energy
          </title>
          <path
            d="M62.0682 7.78728H53.1416V32H41.5539V7.78728H32.6094V0.0179513H62.0682V7.78728ZM92.7523 7.66162V0H66.4833V24.3133C66.4833 28.5103 69.8696 31.9102 74.0418 31.9102H78.9605H92.888V24.2127H79.4035C78.4283 24.2127 77.6353 23.4157 77.6353 22.4355V19.2761H92.145V12.1853H77.6353V7.66162H92.7523ZM19.6178 31.9749H0V0.0502637H19.5535C24.7223 0.0502637 28.9087 4.26164 28.9087 9.45316V22.6366C28.9087 27.7922 24.7509 31.9749 19.6178 31.9749ZM18.1961 21.8826V10.0061C18.1961 8.65253 17.1031 7.5575 15.76 7.5575H11.4699V24.3312H15.76C17.1031 24.3312 18.1961 23.2362 18.1961 21.8826Z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
    <div
      className="dte-wismo-header-link"
    >
      <a
        href="https://www.dteenergy.com"
        target="_self"
      >
        <svg
          className="dte-wismo-hyperlink-icon"
          fill="currentColor"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="external-link"
          >
            <path
              d="M5.75 3C11.1197 3 14 3 14 3M14 3V11.25M14 3L3 14"
              id="Vector"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
          </g>
        </svg>
        Visit DTEEnergy.com
      </a>
    </div>
  </div>
</header>
`;

exports[`<Header /> should render correctly when the current language is es 1`] = `
<header
  className="dte-wismo-header-background"
>
  <div
    className="dte-wismo-header-mobile"
  >
    <button
      className="dte-wismo-header-mobile-hamburger"
      data-testid="open-mobile-menu"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 32 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          menu
        </title>
        <path
          clipRule="evenodd"
          d="M0 1C0 0.447715 0.447715 0 1 0H31C31.5523 0 32 0.447715 32 1C32 1.55228 31.5523 2 31 2H1C0.447715 2 0 1.55228 0 1ZM0 10C0 9.44771 0.447715 9 1 9H31C31.5523 9 32 9.44771 32 10C32 10.5523 31.5523 11 31 11H1C0.447715 11 0 10.5523 0 10ZM0 19C0 18.4477 0.447715 18 1 18H31C31.5523 18 32 18.4477 32 19C32 19.5523 31.5523 20 31 20H1C0.447715 20 0 19.5523 0 19Z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </button>
    <div
      className="dte-wismo-header-mobile-center"
    >
      <div
        className="dte-wismo-header-mobile-logo"
      >
        <a
          href="https://www.header.logo.com"
          title="DTE Energy"
        >
          <svg
            className="dte-wismo-header-logo"
            fill="none"
            height="32"
            viewBox="0 0 93 32"
            width="93"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              DTE Energy
            </title>
            <path
              d="M62.0682 7.78728H53.1416V32H41.5539V7.78728H32.6094V0.0179513H62.0682V7.78728ZM92.7523 7.66162V0H66.4833V24.3133C66.4833 28.5103 69.8696 31.9102 74.0418 31.9102H78.9605H92.888V24.2127H79.4035C78.4283 24.2127 77.6353 23.4157 77.6353 22.4355V19.2761H92.145V12.1853H77.6353V7.66162H92.7523ZM19.6178 31.9749H0V0.0502637H19.5535C24.7223 0.0502637 28.9087 4.26164 28.9087 9.45316V22.6366C28.9087 27.7922 24.7509 31.9749 19.6178 31.9749ZM18.1961 21.8826V10.0061C18.1961 8.65253 17.1031 7.5575 15.76 7.5575H11.4699V24.3312H15.76C17.1031 24.3312 18.1961 23.2362 18.1961 21.8826Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
      <div
        className="dte-wismo-header-page-title"
      >
        Rastreador de Pedidos
      </div>
    </div>
  </div>
  <div
    className="dte-wismo-header"
  >
    <div
      className="dte-wismo-header-page-title"
    >
      Rastreador de Pedidos
    </div>
    <div>
      <a
        href="https://www.header.logo.com"
        title="DTE Energy"
      >
        <svg
          className="dte-wismo-header-logo"
          fill="none"
          height="32"
          viewBox="0 0 93 32"
          width="93"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            DTE Energy
          </title>
          <path
            d="M62.0682 7.78728H53.1416V32H41.5539V7.78728H32.6094V0.0179513H62.0682V7.78728ZM92.7523 7.66162V0H66.4833V24.3133C66.4833 28.5103 69.8696 31.9102 74.0418 31.9102H78.9605H92.888V24.2127H79.4035C78.4283 24.2127 77.6353 23.4157 77.6353 22.4355V19.2761H92.145V12.1853H77.6353V7.66162H92.7523ZM19.6178 31.9749H0V0.0502637H19.5535C24.7223 0.0502637 28.9087 4.26164 28.9087 9.45316V22.6366C28.9087 27.7922 24.7509 31.9749 19.6178 31.9749ZM18.1961 21.8826V10.0061C18.1961 8.65253 17.1031 7.5575 15.76 7.5575H11.4699V24.3312H15.76C17.1031 24.3312 18.1961 23.2362 18.1961 21.8826Z"
            fill="currentColor"
          />
        </svg>
      </a>
    </div>
    <div
      className="dte-wismo-header-link"
    >
      <a
        href="https://www.dteenergy.com"
        target="_self"
      >
        <svg
          className="dte-wismo-hyperlink-icon"
          fill="currentColor"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="external-link"
          >
            <path
              d="M5.75 3C11.1197 3 14 3 14 3M14 3V11.25M14 3L3 14"
              id="Vector"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="1.5"
            />
          </g>
        </svg>
        Visitar DTEEnergy.com
      </a>
    </div>
  </div>
</header>
`;
